bugwarrior/config.py

```diff
@@ -236,14 +236,14 @@ def get_data_path(config, main_section):
     # the `_` subcommands properly (`rc:` can't be used for them).
     line_prefix = 'data.location='
 
-    env={
-        'PATH': os.getenv('PATH'),
-        'TASKRC': taskrc,
-    }
-    # If TASKDATA is set but empty, taskwarrior's data.location is empty.
-    taskdata = os.getenv('TASKDATA')
-    if taskdata:
-        env['TASKDATA'] = taskdata
+    # Take a copy of the environment and add our taskrc to it.
+    env = dict(os.environ)
+    env['TASKRC'] = taskrc
+
+    # Prune the TASKDATA variable if it's completely empty, since otherwise
+    # taskwarrior tries to open the empty-string directory, which fails.
+    if not env.get('TASKDATA',''):
+        del env['TASKDATA']
```

Comment:
I would delete this `TASKDATA` part altogether. This patch changes its nature from working around bugwarrior-caused misconfiguration to working around user-caused misconfiguration.

The reason we needed to have this conditional before is that `os.getenv('TASKDATA')` returns an empty string when the `TASKDATA` environment variable *is not set*.

After your revision, its purpose is to silently work around a mis-configured system in which the user has assigned `TASKDATA` an empty string. I don't think this is going to be a very frequent scenario. If it were, a more appropriate response would be to raise a warning or an exception. Also bugwarrior would probably break down at a lot of other points if `TASKDATA` were set to empty since this is far from our only invocation of `task`.